core/src/main/java/com/graphhopper/util/PathSimplification.java

```diff
@@ -98,69 +94,57 @@ public PointList simplify() {
             if (toSimplifyIndex >= 0 && simplificationPossible) {
                 // Only simplify if there is more than one point
                 if (nonConflictingEnd - nonConflictingStart > 1) {
-                    // Simplify
                     int removed = douglasPeucker.simplify(pointList, nonConflictingStart, nonConflictingEnd);
                     if (removed > 0) {
                         for (int i = 0; i < toSimplify.size(); i++) {
-                            reduceNumberOfPoints(toSimplify.get(i), offset[i], removed, startIntervals[i], endIntervals[i] - removed);
```

Comment:
The nonConflictingXY describes the index from where to where we can simplify without potentially remove a point that we might have to reference later.

> I think we have to call this for all PathDetails even if no simplification is possible, as we need to change the pathDetail.first variable for all entries following a simplified PathDetail.

Yes this is true, since we don't store the length anymore, but the indexes we have to propagate our change trough all PathDetails, which is not very nice. That's btw the main reason why I decided earlier to only use length instead of indexes.